solr/core/src/java/org/apache/solr/query/SolrRangeQuery.java

```diff
@@ -452,6 +455,7 @@ private SegState getSegState(LeafReaderContext context) throws IOException {
 
       if (doCheck) {
         answer = createDocSet(solrSearcher, count);
+        // This can be a naked put because the cache usually gets checked in SolrIndexSearcher
```

Comment:
A naked put in this context is a put without a previous call to get or without a computeIfAbsent. We typically don't want to just insert things into the cache, but in this case it's _probably_ ok because we would have already checked in SIS if the entry exists.